src/DrTests-CommentsToTests/DTCommentTestConfiguration.class.st

```diff
@@ -9,17 +9,14 @@ Class {
 
 { #category : #converting }
 DTCommentTestConfiguration >> asTestSuite [
-	| suite |
+	| suite methods |
 	suite := TestSuite new.
 	suite := TestSuite named: 'Test Generated From Comments'.
-	self items
-		reject: [ :each | each isAbstract ]
-		thenDo: [ :c | 
-			c methods
-				do: [ :m | 
-					(m methodClass compiledMethodAt: m selector) comments
-						do: [ :com | 
-							(com includesSubstring: '>>>')
-								ifTrue: [ suite addTest: (CommentTestCase comment: com class: c selector: m selector) ] ] ] ].
+	methods := (self items reject: [ :each | each isAbstract ]) flatCollect: [ :each | each methods ].
+	methods := methods select: [:each | each sourceCode includesSubstring: '>>>' ].
```

Comment:
That was quite ugly code. We should refactor it further to have good API:
- aMethod hasExecutableComment
-  aMethod collectExecutableComments.

Reply:
yes, the idea was to do just the tiniest local change... there is an api to get the example objects that should be used here